### ISSUE TITLE
MQTT Error when listening for messages

### ISSUE THREAD

**Issue by Strafos** [carpedm20/fbchat#495]
## Description of the problem
When I try to listen to messages, I get this every few seconds:
```
MQTT Error: A network protocol error occurred when communicating with the broker.
Got exception while listening
NoneType: None
```

## Code to reproduce
echobot

## Environment information
- python3.7.5
- `fbchat` version 1.9.1

**Comment by dundalkian** [comment]
I am having this problem with listen() as well. With my old working 1.8 code as well as the echobot example.

Are there any new steps to either logging in/giving credentials or listening that have been made necessary by the switch to websockets?

**Comment by xaadu** [comment]
I'm facing same problem!

**Comment by madsmtm** [comment]
Thanks for the report, this is not an error I've experienced myself, so I'll need your help fixing it.

Can you enable debug output by typing the following in front of your code:
```python
import logging

logging.basicConfig(level=logging.DEBUG)
```

And then giving me the output (after having removed sensitive data)?

**Comment by HunterAzar** [comment]
Login of ####### successful.
INFO:client:Login of ########
successful.
Listening...
INFO:client:Listening...
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): 0-edge-chat.facebook.com:443
DEBUG:urllib3.connectionpool:https://0-edge-chat.facebook.com:443 "GET /active_ping?seq=0&channel=p_100007839900545&clientid=32afe5f7&partition=-2&cap=0&uid=100007839900545&viewer_uid=100007839900545&state=active&__a=1&__req=1&__rev=1001582455&fb_dtsg=AQH3VDLGHmLS%3AAQFnuyRa1_qo HTTP/1.1" 200 48
DEBUG:client:{'t': 'pong'}
DEBUG:urllib3.connectionpool:https://0-edge-chat.facebook.com:443 "GET /pull?seq=0&msgs_recv=0&clientid=32afe5f7&state=active&__a=1&__req=2&__rev=1001582455&fb_dtsg=AQH3VDLGHmLS%3AAQFnuyRa1_qo HTTP/1.1" 200 105
DEBUG:client:{'t': 'lb', 'lb_info': {'sticky': '975', 'pool': 'rfrc0c01_chatproxy-regional'}}
DEBUG:urllib3.connectionpool:https://0-edge-chat.facebook.com:443 "GET /active_ping?seq=0&channel=p_100007839900545&clientid=32afe5f7&partition=-2&cap=0&uid=100007839900545&sticky_token=975&sticky_pool=rfrc0c01_chatproxy-regional&viewer_uid=100007839900545&state=active&__a=1&__req=3&__rev=1001582455&fb_dtsg=AQH3VDLGHmLS%3AAQFnuyRa1_qo HTTP/1.1" 200 48
DEBUG:client:{'t': 'pong'}
DEBUG:urllib3.connectionpool:https://0-edge-chat.facebook.com:443 "GET /pull?seq=0&msgs_recv=0&sticky_token=975&sticky_pool=rfrc0c01_chatproxy-regional&clientid=32afe5f7&state=active&__a=1&__req=4&__rev=1001582455&fb_dtsg=AQH3VDLGHmLS%3AAQFnuyRa1_qo HTTP/1.1" 200 123
DEBUG:client:{'t': 'fullReload', 'seq': 1, 'u': 100007839900545, 'ms': [{'type': 'qprimer', 'made': '1578417901412'}]}
DEBUG:urllib3.connectionpool:https://0-edge-chat.facebook.com:443 "GET /active_ping?seq=1&channel=p_100007839900545&clientid=32afe5f7&partition=-2&cap=0&uid=100007839900545&sticky_token=975&sticky_pool=rfrc0c01_chatproxy-regional&viewer_uid=100007839900545&state=active&__a=1&__req=5&__rev=1001582455&fb_dtsg=AQH3VDLGHmLS%3AAQFnuyRa1_qo HTTP/1.1" 200 48
DEBUG:client:{'t': 'pong'}
DEBUG:urllib3.connectionpool:https://0-edge-chat.facebook.com:443 "GET /pull?seq=1&msgs_recv=0&sticky_token=975&sticky_pool=rfrc0c01_chatproxy-regional&clientid=32afe5f7&state=active&__a=1&__req=6&__rev=1001582455&fb_dtsg=AQH3VDLGHmLS%3AAQFnuyRa1_qo HTTP/1.1" 200 54
DEBUG:client:{'t': 'heartbeat'}
DEBUG:urllib3.connectionpool:https://0-edge-chat.facebook.com:443 "GET /active_ping?seq=0&channel=p_100007839900545&clientid=32afe5f7&partition=-2&cap=0&uid=100007839900545&sticky_token=975&sticky_pool=rfrc0c01_chatproxy-regional&viewer_uid=100007839900545&state=active&__a=1&__req=7&__rev=1001582455&fb_dtsg=AQH3VDLGHmLS%3AAQFnuyRa1_qo HTTP/1.1" 200 48
DEBUG:client:{'t': 'pong'}
DEBUG:urllib3.connectionpool:https://0-edge-chat.facebook.com:443 "GET /pull?seq=0&msgs_recv=0&sticky_token=975&sticky_pool=rfrc0c01_chatproxy-regional&clientid=32afe5f7&state=active&__a=1&__req=8&__rev=1001582455&fb_dtsg=AQH3VDLGHmLS%3AAQFnuyRa1_qo HTTP/1.1" 200 115
DEBUG:client:{'t': 'msg', 'seq': 1, 'u': 100007839900545, 'ms': [{'type': 'qprimer', 'made': '1578417901412'}]}
DEBUG:urllib3.connectionpool:https://0-edge-chat.facebook.com:443 "GET /active_ping?seq=1&channel=p_100007839900545&clientid=32afe5f7&partition=-2&cap=0&uid=100007839900545&sticky_token=975&sticky_pool=rfrc0c01_chatproxy-regional&viewer_uid=100007839900545&state=active&__a=1&__req=9&__rev=1001582455&fb_dtsg=AQH3VDLGHmLS%3AAQFnuyRa1_qo HTTP/1.1" 200 48
DEBUG:client:{'t': 'pong'}

**Comment by HunterAzar** [comment]
https://pastebin.com/1g3MV7fR

**Comment by madsmtm** [comment]
Hi @HunterAzar, thanks for the dump, but I think your problem is different from the one in this issue. Are you sure you've installed the latest `v1.9.1` version of `fbchat`?

**Comment by Strafos** [comment]
https://pastebin.com/92XuCWcx
I took out some of the user id fields

**Comment by JarbasAl** [comment]
i am having exactly same issue

potentially useful info: 
- connecting from Portugal
- newly created account
- installed by pip
- v1.9.1
- echobot example
- xubuntu

not pasting logs because i'm getting same output as @Strafos

**Comment by gave92** [comment]
Issue seems to be that no cookies are passed when creating the mqtt websocket

**Comment by madsmtm** [comment]
Thanks for the data @Strafos and @JarbasAI, that helped identifying that the problem was in the connection (we received an invalid `CONNACK` code).

And thanks @gave92, that fixes the problem, my mistake was that I was testing on cookies loaded from file, which didn't have the same parameters as cookies received from Facebook.

I've pushed a fix in fb63ff0db86f6b3118b5f1b65f3f9f462afb5c10, and released `v1.9.2`, thanks for your patience 👍

**Comment by h4o** [comment]
I just upgraded but have a new issue
```
Listening...
INFO:client:Listening...
DEBUG:paho.mqtt.client:Received CONNACK (0, 0)
DEBUG:paho.mqtt.client:Sending PUBLISH (d0, q1, r0, m1), 'b'/messenger_sync_create_queue'', ... (188 bytes)
DEBUG:paho.mqtt.client:Received PUBACK (Mid: 1)
DEBUG:paho.mqtt.client:Received PUBLISH (d0, q1, r1, m6190), '/t_ms', ...  (74 bytes)
DEBUG:paho.mqtt.client:Sending PUBACK (Mid: 6190)
ERROR:paho.mqtt.client:Caught exception in on_message: the JSON object must be str, not 'bytes'
DEBUG:paho.mqtt.client:Received PUBLISH (d0, q0, r1, m0), '/orca_presence', ...  (913 bytes)
ERROR:paho.mqtt.client:Caught exception in on_message: the JSON object must be str, not 'bytes'
DEBUG:paho.mqtt.client:Sending PINGREQ
DEBUG:paho.mqtt.client:Received PINGRESP
DEBUG:paho.mqtt.client:Sending PINGREQ
DEBUG:paho.mqtt.client:Received PINGRESP
DEBUG:paho.mqtt.client:Received PUBLISH (d0, q0, r1, m0), '/orca_typing_notifications', ...  (49 bytes)
ERROR:paho.mqtt.client:Caught exception in on_message: the JSON object must be str, not 'bytes'
DEBUG:paho.mqtt.client:Received PUBLISH (d0, q0, r1, m0), '/orca_presence', ...  (92 bytes)
ERROR:paho.mqtt.client:Caught exception in on_message: the JSON object must be str, not 'bytes'
DEBUG:paho.mqtt.client:Received PUBLISH (d0, q1, r1, m6194), '/t_ms', ...  (649 bytes)
DEBUG:paho.mqtt.client:Sending PUBACK (Mid: 6194)
ERROR:paho.mqtt.client:Caught exception in on_message: the JSON object must be str, not 'bytes'
DEBUG:paho.mqtt.client:Received PUBLISH (d0, q0, r1, m0), '/orca_typing_notifications', ...  (49 bytes)
ERROR:paho.mqtt.client:Caught exception in on_message: the JSON object must be str, not 'bytes'
````
Looks like a message format error (or an encoding issue?)

**Comment by madsmtm** [comment]
Thanks @h4o, this was an issue in Python 3.4/3.5, that I didn't know about: Apparently `json.loads` transparently loads `bytes` and `str` in Python 2.7/3.6/3.7, but not in 3.0-3.5...

Fixed in 4714be5697b665c531d230b9190c533c57938e9e, release `v1.9.3`
